integration_test/metadata/integration_metadata.go

```diff
@@ -209,31 +233,35 @@ func AssertMetric(metric *ExpectedMetric, series *monitoringpb.TimeSeries) error
 }
 
 func assertMetricLabels(metric *ExpectedMetric, series *monitoringpb.TimeSeries) error {
-	// Only expected labels must be present
 	var err error
+	// Only expected labels must be present
+	expectedLabels := make(map[string]bool)
+	for _, expectedLabel := range metric.Labels {
+		expectedLabels[expectedLabel.Name] = true
+	}
 	for actualLabel, actualValue := range series.Metric.Labels {
-		if _, ok := metric.Labels[actualLabel]; !ok {
+		if _, ok := expectedLabels[actualLabel]; !ok {
```

Comment:
Since we're only populating `expectedLabels` with `true` values:

```suggestion
		if !expectedLabels[actualLabel]; {
```

Reply:
Sigh, I typo'd the suggestion -- that trailing semicolon shouldn't be there.

Reply:
Fixed.